src/pages/QuestionPage.tsx

```diff
@@ -103,17 +112,39 @@ export default function QuestionPage() {
     }
   }
 
-  const nextQuestion = (event:React.MouseEvent) => {
+  const nextQuestion = (event: React.MouseEvent) => {
     // Submit answer to backend
-    if (!requestNextQuestionButtonPressed){
+    if (!requestNextQuestionButtonPressed) {
       console.log("Next question button pressed");
       dispatch(setRequestNextQuestion(true));
       setRequestNextQuestionButtonPressed(true);
+      // reset the leaderboard for the next question
+      dispatch(resetLeaderboard());
+      dispatch(setRequestUpdatedLeaderboard(false));
+      setShowLeaderboardButtonClicked(false);
+    }
+  }
+
+  const showQuestion = (event: React.MouseEvent) => {
+    if (showLeaderboard) {
+      setShowLeaderboardButtonClicked(false);
+      setShowLeaderboardButtonClicked(false);
+    }
+  }
+
+  const getLeaderboard = (event: React.MouseEvent) => {
+    // Submit answer to backend
+    // reset the leaderboard if it has not already been reset
+    dispatch(resetLeaderboard());
+    if (!showLeaderboard) {
```

Comment:
Might be a good idea to remove this flag. There's a bug that exists where if you press on the button again it'll clear the leaderboard (i.e. all the names and scores are removed from the leaderboard)